controller/controller_server/csi_controller_server.py

```diff
@@ -512,14 +512,14 @@ def DeleteSnapshot(self, request, context):
                 except controller_errors.ObjectNotFoundError as ex:
                     logger.debug("Snapshot was not found during deletion: {0}".format(ex))
 
-        except controller_errors.ObjectNotFoundError:
-            logger.debug("snapshot was not found during deletion: {0}".format(ex))
+        except controller_errors.ObjectNotFoundError as ex:
+            logger.debug("snapshot was not found during deletion: {0}".format(ex.message))
```

Comment:
Why do we format the `ex.message` instead of `ex`? Some exceptions don't have meaningful messages

Reply:
could you please [point to an exception](https://github.com/IBM/ibm-block-csi-driver/blob/73e76506dee47c5fcc8f4069931db994af3087b7/controller/array_action/errors.py) which doesn't have a meaningful message?
I think meaningful messages are important

Reply:
I meant non-IBM exceptions (Pythonic exceptions, 3rd Parties exceptions, etc.).
Personally, in IBMic exceptions, I prefer to override `__str__` and/or `__repr__` and return the relevant information because it saves the developer checking if the exception has a meaningful exception or not.

Reply:
I am not sure I follow - AFAIK these formattings of `ex.message` are done only for our own exceptions, so I don't see an issue with this change.
regarding the refactor you proposed - it sounds possible, but not related to this PR, feel free to open a ticket for it.

Reply:
A [ticket](https://jira.xiv.ibm.com/browse/CSI-2785) has been created.
Let's continue discussing there.

Reply:
please avoid using internal links in public repos